polaris-react/src/components/DataTable/DataTable.tsx

```diff
@@ -504,15 +507,16 @@ class DataTableInner extends PureComponent<CombinedProps, DataTableState> {
     button.style.removeProperty('visibility');
   };
 
-  private calculateColumnVisibilityData = (condensed: boolean) => {
+  private calculateColumnVisibilityData = (condensed?: boolean) => {
     const fixedFirstColumns = this.fixedFirstColumns();
     const {
       table: {current: table},
       scrollContainer: {current: scrollContainer},
       dataTable: {current: dataTable},
     } = this;
+    const {stickyHeader} = this.props;
 
-    if (condensed && table && scrollContainer && dataTable) {
+    if ((stickyHeader || condensed) && table && scrollContainer && dataTable) {
```

Comment:
`stickyHeader` also needs the measurements to correctly apply the widths to it's headings.